### ISSUE TITLE
After updating to v 5.5.3 and above the frontend breaks

### ISSUE THREAD

**Issue by firmenpunkt** [tpwd/ke_search#257]
TYPO3 12.4
No starting point found. Please set the starting point in the plugin configuration or via TypoScript: https://docs.typo3.org/p/tpwd/ke_search/main/en-us/Configuration/OverrideRecordStoragePage.html.
What is the reason?

**Comment by firmenpunkt** [comment]
the problem was an additional TS inclusion, some old old code:

########################################
# Searchbox Plugin
########################################
lib.searchbox = COA_INT
lib.searchbox {
  #10 < plugin.tx_kesearch_pi1

  # result page
  #10.resultPage = 174
  
  commented out, sry to use your time, I will close now

**Comment by firmenpunkt** [comment]
Sorry I think I solved it, but no. If I restore the files
[Classes/Lib/PluginBaseHelper.php]
and
[Classes/Lib/Db.php]
from v 5.5.2., it works in all versions.

**Comment by firmenpunkt** [comment]
Is it possible that this is s TYPO3 12.4.22 problem? The CType is wrong if I update from 5.4.2 to 6.0.1.
If I try to configure ke_search 6.0.1 for instance, there is no plugin "Faceted Search: Show searchbox and filters", only Searchbox. And it moved to the General tab. If I choose Searchbox, I can only set the Startingpoint, but not the Record storage, where the indexer is configured. In the plugin Search resultlist, I must set the Startingpoint again. There is no possibily to choose the folder with the indexer/record storage in either of the plugins.

**Comment by christianbltr** [comment]
The CType has changed in v6.0.0. That's mentioned in the ChangeLog. That's according to https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/13.4/Deprecation-105076-PluginContentElementAndPluginSubTypes.html#migration

Please make sure to run the upgrade wizard ke_search provides.

In 5.4.2 and 6.0.0 there was a problem which cause the "Startinpoint" field not to be shown. That has been fixed in 6.0.1 and 5.4.3.

Does upgrading to one of these versions (and running the upgrade wizard if you choose to use version 6) help?

**Comment by firmenpunkt** [comment]
My problem with the configuration remains. Unlike in prevoius versions in the newer ones I cannot select the storage where the indexer and the filters are configured. Rather I can select the starting point (of the pages to be indexed I guess) in both plugins: searchbox and search search results. The upgrade wizard says 100% as and the table compare is also on "good job!". And the frontend says I haven't selected the startingpont, no matter what I select in the plugin.

**Comment by maikschneider** [comment]
Hey, I encountered the same problem with TYPO3 v11, ke_search 5.5.6:
Suddenly it is required to define startingPoints for plugins (without flexform), this was not needed in previous versions.
My fix:
```diff
lib.quickSearch = COA
lib.quickSearch {
  10 < plugin.tx_kesearch_pi3
  10 {
    templateLayout = 20
    resultPage = 111
+    overrideStartingPoint = 110
  }
}
```

**Comment by firmenpunkt** [comment]
> ```diff
> overrideStartingPoint = 110
> ```
cool thanks, this "hack" works for me!

**Comment by firmenpunkt** [comment]
but in the versions > 6 the ctype changes, and one cannot configure the indexer (you cannot select the storage folder where it is configured in the plugin) an therefore the indexing and the search does not work anymore:

Faceted Search
Indexer is idle.
The index is empty.
No indexer configurations have been found. Please create an indexer configuration record first!

is there a fix for that?

**Comment by christianbltr** [comment]
The check for an existing starting point configuration has now been moved to a later point where the starting point is really necessary. This fixes the regression which required a starting point to be always set.

The fix will be in versions 5.6.1 and 6.1.2.

@firmenpunkt I'm not sure that your comment https://github.com/tpwd/ke_search/issues/257#issuecomment-2470149080 relates to the same issue because as far as I can see you are referring to the backend configuration while the bug in this issue was related to the frontend. May I ask you to open a new issue and provide some screenshots?
